pkg/secretless/entrypoint/start_secretless.go

```diff
@@ -0,0 +1,124 @@
+package entrypoint
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/cyberark/secretless-broker/internal"
+	secretlessLog "github.com/cyberark/secretless-broker/internal/log"
+	"github.com/cyberark/secretless-broker/internal/plugin"
+	"github.com/cyberark/secretless-broker/internal/plugin/v1/event_notifier"
+	"github.com/cyberark/secretless-broker/internal/profile"
+	"github.com/cyberark/secretless-broker/internal/signal"
+	"github.com/cyberark/secretless-broker/pkg/secretless"
+	"github.com/cyberark/secretless-broker/pkg/secretless/config"
+	v2 "github.com/cyberark/secretless-broker/pkg/secretless/config/v2"
+)
+
+// CLIParams holds the command line flag information that Secretless was started
+// with.
+type CLIParams struct {
+	ConfigFile          string
+	ConfigManagerSpec   string
+	DebugEnabled        bool
+	FsWatchEnabled      bool
+	PluginChecksumsFile string
+	PluginDir           string
+	ProfilingMode       string
+	ShowVersion         bool
+}
+
+// StartSecretless method is the main entry point into the broker after the CLI
+// flags have been parsed
+func StartSecretless(params *CLIParams) {
+
+	ShowVersion(params.ShowVersion)
+
+	VerifyPlugins(params.PluginDir, params.PluginChecksumsFile)
+
+	// Construct the deps of Secretless
+	cfg := ReadConfig(params.ConfigFile)
+	logger := secretlessLog.New(params.DebugEnabled)
+	evtNotifier := event_notifier.New(nil)
+	availPlugins := &internal.AvailPluginStub{}
+
+	// Prepare Secretless
+	secretless := internal.NewSecretless(cfg, availPlugins, logger, evtNotifier)
+	signal.StopOnExitSignal(secretless)
+
+	HandlePerformanceProfiling(params.ProfilingMode)
+
+	secretless.Start()
+}
+
+func ReadConfig(cfgFile string) v2.Config {
+	// TODO: Add back in CRD / generalized config option
+	cfg, err := config.LoadFromFile(cfgFile)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	return cfg
+}
+
+func ShowVersion(showVersionAndExit bool) {
```

Comment:
nit: we're already in the ShowVersion method so that part of the param name can most likely be removed since it's somewhat redundant

Reply:
nit: This method (since it's public) is lacking a comment that golint/govet might complain about